dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -198,9 +200,29 @@ private static void ProcessPositionalArguments(KernelFunctionMetadata functionMe
     private static object? ParseResult(FunctionResult result)
     {
         var resultAsObject = result.GetValue<object?>();
-        if (result.ValueType is not null && resultAsObject is not null && result.ValueType == typeof(OpenAIChatMessageContent))
+        if (result.ValueType is not null && resultAsObject is not null)
         {
-            resultAsObject = ((OpenAIChatMessageContent)resultAsObject).Content;
+#pragma warning disable SKEXP0042 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
```

Comment:
Huh, are we making use of an experimental feature?  If so, shouldn't we be experimental too?

Reply:
Yeah, working it out with Dmytro. Closing this PR as a dupe so will continue here: https://github.com/microsoft/semantic-kernel/pull/4343/files